File: 2021-22/contest1/prog/intermediate/fibonacci_clock.py
import os
import sys

for n in range(5): # This loop is not needed on hackerrank.com!
	lower_one, upper_one, two, three, five = [x for x in input().split()]
	hh = 0
	mm = 0
	# red
	if lower_one == 'R':
		hh += 1
	if upper_one == 'R':
		hh += 1
	if two == 'R':
		hh += 2
	if three == 'R':
		hh += 3
	if five == 'R':
		hh += 5
	# green
	if lower_one == 'G':
		mm += 1
	if upper_one == 'G':
		mm += 1
	if two == 'G':
		mm += 2
	if three == 'G':
		mm += 3
	if five == 'G':
		mm += 5
	# blue
	if lower_one == 'B':
		hh += 1
		mm += 1
	if upper_one == 'B':
		hh += 1
		mm += 1;
	if two == 'B':
		hh += 2
		mm += 2
	if three == 'B':
		hh += 3
		mm += 3
	if five == 'B':
		hh += 5
		mm += 5
	mm *= 5
	if mm >= 60:
		hh += (mm//60)
		mm %= 60
	if hh > 11:
		hh %= 12
	print("{0:02}:{1:02}".format(hh,mm))

File: 2021-22/contest1/prog/intermediate/fibonacci_clock.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	for(int n = 0; n < 5; ++n) { // This loop is not needed on hackerrank.com!
		char lower_one, upper_one, two, three, five;
		cin >> lower_one >> upper_one >> two >> three >> five;
		int hour = 0, min = 0;
		// red
		if(lower_one == 'R') hour += 1;
		if(upper_one == 'R') hour += 1;
		if(two == 'R') hour += 2;
		if(three == 'R') hour += 3;
		if(five == 'R') hour += 5;
		// green
		if(lower_one == 'G') min += 1;
		if(upper_one == 'G') min += 1;
		if(two == 'G') min += 2;
		if(three == 'G') min += 3;
		if(five == 'G') min += 5;
		// blue
		if(lower_one == 'B') hour += 1, min += 1;
		if(upper_one == 'B') hour += 1, min += 1;
		if(two == 'B') hour += 2, min += 2;
		if(three == 'B') hour += 3, min += 3;
		if(five == 'B') hour += 5, min += 5;
		min *= 5;
		if(min >= 60) hour += (min/60), min %= 60;
		if(hour > 11) hour %= 12;
		cout<< setw(2) << setfill('0') << hour << ':'
			<< setw(2) << setfill('0') << min << endl;
	}
	return 0;
}
